### PR TITLE
Docker sim cherry pick

### PR DESCRIPTION
**Pull request by MrMondrian**
## Related Issue(s)
  - https://github.com/mcgill-robotics/AUV-2024/issues/241
  - https://github.com/mcgill-robotics/AUV-2024/issues/269

## Other Dependent PRs
  - https://github.com/mcgill-robotics/auv-sim-gazebo/pull/41

## Changes Made
- Adds rostest case for github actions that runs in gazebo

## Rationale/Other Solutions Considered
- Automates integration test
- Don't need to nag new members if they tested in sim
- Will be very easy to expand this for more tests

## Tests Done
- Tested locally using act, need to test on the main branch unfortunately

## Required Wiki Changes (if any)
- Might be smart to add a page for all actions

### CODE REVIEW

File: catkin_ws/src/planner/src/testMission.py
Comment (antoinedang) [inline catkin_ws/src/planner/src/testMission.py:1]: can we rename these files (test.launch, testMission.launch. testMission.py) into something clearer? Its no obvious from the name that the test is for continuous integration tests

File: catkin_ws/src/planner/src/testMission.py [hunk 1]
```diff
@@ -2,11 +2,36 @@
...
 
-if __name__ == "__main__":
-    rospy.init_node("moveForward",log_level=rospy.DEBUG)
-    controls = Controller(rospy.Time(0))
-    controls.move((0,0,0))
+ 
+## A sample python unit test
+class Test1(unittest.TestCase):
+    ## test 1 == 1
+    def test__simple_movement(self): # only functions with 'test_'-prefix will be run!
```
Comment (antoinedang) [inline catkin_ws/src/planner/src/testMission.py]: while we're at it lets test some more complex orientations/position changes (can be done in a further PR, maybe lets make an issue out of it if we want to give this to a new member)

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -14,6 +14,13 @@ jobs:
       - name: catkin build
         shell: bash
         run: |
+            git submodule update --init --recursive
             source /opt/ros/noetic/setup.bash
             cd catkin_ws
             catkin build
+      - name: simple test
```
Comment (antoinedang) [inline .github/workflows/pull_request.yml]: can we rename the test from "simple test" to something like "sim test" or "planner test" or something so its clear when it fails what its testing

**Review by antoinedang — CHANGES_REQUESTED** [review]
commented

**Comment by antoinedang** [comment]
i forgot but also because of adding the sim as a git submodule we should update the setup instructions in the wiki

File: .github/workflows/pull_request.yml (hunk 2 above)
Comment (MrMondrian) [reply .github/workflows/pull_request.yml]: fixed

File: catkin_ws/src/planner/src/testMission.py (hunk 1 above)
Comment (MrMondrian) [reply catkin_ws/src/planner/src/testMission.py]: lets save this for a new member

File: catkin_ws/src/planner/src/testMission.py
Comment (MrMondrian) [reply catkin_ws/src/planner/src/testMission.py:1]: done

**Review by antoinedang — APPROVED** [review]

**Review by frompotenza — APPROVED** [review]

**Pull request merged**